ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,393 @@
+
+Here is the initial schema for our application. We'll be updating it slightly later in this guide, so don't run it quite yet.
+
+::
+
+  -- Don't try executing this schema as-is, we need to make
+  -- a few changes later on to prepare it for Citus
+
+  CREATE TABLE companies (
+    id bigserial PRIMARY KEY,
+    name text NOT NULL,
+    image_url text,
+    created_at timestamp without time zone NOT NULL,
+    updated_at timestamp without time zone NOT NULL
+  );
+
+  CREATE TABLE campaigns (
+    id bigserial PRIMARY KEY,
+    company_id bigint REFERENCES companies (id),
+    name text NOT NULL,
+    cost_model text NOT NULL,
+    state text NOT NULL,
+    monthly_budget bigint,
+    blacklisted_site_urls text[],
+    created_at timestamp without time zone NOT NULL,
+    updated_at timestamp without time zone NOT NULL
+  );
+
+  CREATE TABLE ads (
+    id bigserial PRIMARY KEY,
+    campaign_id bigint REFERENCES campaigns (id),
+    name text NOT NULL,
+    image_url text,
+    target_url text,
+    impressions_count bigint DEFAULT 0,
+    clicks_count bigint DEFAULT 0,
+    created_at timestamp without time zone NOT NULL,
+    updated_at timestamp without time zone NOT NULL
+  );
+
+  CREATE TABLE clicks (
+    id bigserial PRIMARY KEY,
+    ad_id bigint REFERENCES ads (id),
+    clicked_at timestamp without time zone NOT NULL,
+    site_url text NOT NULL,
+    cost_per_click_usd numeric(20,10),
+    user_ip inet NOT NULL,
+    user_data jsonb NOT NULL
+  );
+
+  CREATE TABLE impressions (
+    id bigserial PRIMARY KEY,
+    ad_id bigint REFERENCES ads (id),
+    seen_at timestamp without time zone NOT NULL,
+    site_url text NOT NULL,
+    cost_per_impression_usd numeric(20,10),
+    user_ip inet NOT NULL,
+    user_data jsonb NOT NULL
+  );
+
+This schema supports querying the performance of ads and campaigns. It is designed for a single-machine database, and will require adjustment in a distributed environment. To see why, we must become familar with how Citus distributes data and executes queries.
+
+Applications connect to a certain PostgreSQL server in the Citus cluster called the *coordinator node.* The connection is established using an ordinary PostgreSQL `connection URI <https://www.postgresql.org/docs/current/static/libpq-connect.html#AEN45527>`_. However the actual data and processing is stored on and will happen in other machines called *worker nodes.*
+
+The coordinator examines each client query and determines what data the query needs, and which worker nodes have the data. The coordinator then splits the query into simplified *query fragments*, and sends them to worker nodes for processing. When the workers' results are ready, the coordinator puts them together into a final result and forwards it to the application.
```

Comment:
Could we alter this description to fir better with MT workloads? This is a description of the real-time executor i.e. splits into query fragments, co-ordinator puts together the results etc., however Multi tenancy mostly uses router.

Could we say something like "data is send to multiple nodes, and when queries come in, Citus figures out the right node to send the query to and fetches the results and returns them...."